### DISCUSSION TITLE
procedural tiles

### DISCUSSION THREAD

**Discussion by alienatorZ** [iTowns/itowns#1942] [Q&A] [answered]
How would I create a texture, say on a canvas, then replicate that texture for tiles all over the globe? Also how would I display the tiles at 0 elevation and none at higher elevations?

Thank you

**Comment by mgermerie** [comment, marked as answer]
Hi ! You can see [here](https://github.com/mgermerie/itowns/blob/misc/tms-image/examples/test.html) a small example I made to display an image on each tile over the globe. You can see the result on the screenshot below, where each red rectangle is a globe tile.
![Capture d’écran du 2022-12-15 16-51-31](https://user-images.githubusercontent.com/73115044/207906361-20347bc0-263d-40b6-8243-f0a4fda3905f.png)

You could draw on a canvas, and pass `canvas.toDataURL()` as the `url` of the `TMSSource`. Is it what you need ?

@gchoqueux, perhaps do you know of another way of doing so ?

About displaying tiles and elevation, I am not sure I understood what you want. Do you want to avoid displaying tiles which have non null elevation data attached ?

**Reply by gchoqueux** [reply]
@alienatorZ could specify your requests?

**Comment by alienatorZ** [comment]
Thank you for that example it helps greatly!  For the elevation question I was trying to display contour like colors and be able to adjust them dynamically according to elevation.
